posthog/hogql/placeholders.py

```diff
@@ -23,24 +23,35 @@ def __init__(self):
     def visit_cte(self, node: ast.CTE):
         super().visit(node.expr)
 
-    def visit_placeholder(self, node: ast.Placeholder):
-        self.found.add(node.field)
+    def visit_block(self, node: ast.Block):
+        # TODO: remove all this when we can use bytecode in placeholders
+        if len(node.declarations) > 1:
+            raise QueryError("Placeholders can only contain a single declaration")
```

Comment:
Nit: "Blocks can only contain a single declaration"

Reply:
I'm not a fan of the language mix of "block" and "placeholder" - especially with removing `ast.Placeholder` node. I'd be in favour of removing _all_ `placeholder` references in favour of "block" here